clients/admin-ui/src/features/datamap/datamap.slice.ts

```diff
@@ -104,6 +107,60 @@ const datamapApi = baseApi.injectEndpoints({
       },
       providesTags: ["Datamap"],
     }),
+    exportMinimalDatamapReport: build.mutation<
+      Page_DatamapReport_,
+      {
+        groupBy: DATAMAP_GROUPING;
+        pageIndex: number;
+        pageSize: number;
+        search: string;
+        dataUses?: string;
+        dataCategories?: string;
+        dataSubjects?: string;
+        format?: ExportFormat;
+      }
+    >({
+      query: ({
+        groupBy,
+        pageIndex,
+        pageSize,
+        search,
+        dataUses,
+        dataCategories,
+        dataSubjects,
+        format,
+      }) => {
+        let queryString = `page=${pageIndex}&size=${pageSize}&group_by=${groupBy}`;
+        if (dataUses) {
+          queryString += `&${dataUses}`;
+        }
+        if (dataCategories) {
+          queryString += `&${dataCategories}`;
+        }
+        if (dataSubjects) {
+          queryString += `&${dataSubjects}`;
+        }
+        if (search) {
+          queryString += `&search=${search}`;
+        }
+        return {
+          url: `plus/datamap/minimal/${format}?${queryString}`,
+          responseHandler: async (response) => {
+            const filename = await getFileNameFromContentDisposition(
+              response.headers.get("content-disposition")
+            );
+            const arrayBuffer = await response.arrayBuffer();
+            const blob = new Blob([arrayBuffer], {
+              type:
+                response.headers.get("content-type") ||
+                "application/octet-stream",
+            });
+            saveAs(blob, filename);
+            return { data: undefined }; // once the file is downloaded, we no longer want this to be cached in the browser so we return undefined to RTK Query
```

Comment:
Out of curiosity, can you explain more about how browser caching works here by default and what RTK is?

Reply:
All of these slices are using our base `api.slice.ts` which is ultimately using `@reduxjs/toolkit/query` (aka. RTK Query) whose primary function is to maintain responses in browser cache for UI state management (where we get our `isLoading` etc. from). Since this data is not ever going to the UI, it's pointless to even involve that cache management functionality. But, it would be a big pain to write a custom fetch that does all of the authentication and everything provided from the RTK Query config so I'm still using it here.